contrib/scripts/check-k8s-code-gen.sh

```diff
@@ -22,7 +22,7 @@ make generate-k8s-api manifests
 git add --intent-to-add .
 
 # Check for diff
-diff="$(git diff)"
+diff="$(git diff --staged)"
```

Comment:
we should add this here too: https://github.com/cilium/cilium/blob/main/contrib/scripts/check-api-code-gen.sh#L41